docs/reference/reranking/index.asciidoc

```diff
@@ -0,0 +1,71 @@
+[[re-ranking-overview]]
+= Re-ranking
+
+Many search systems are built on two-stage retrieval pipelines.
+
+The first stage uses cheap, fast algorithms to find a subset of possible matches.
+
+The second stage uses a more powerful model, often machine learning-based, to reorder the documents.
+This second step is called re-ranking.
+This approach balances computational costs, because the resource-intensive model is only applied to a smaller set of pre-filtered results.
```

Comment:
```suggestion
Because the resource-intensive model is only applied to the smaller set of pre-filtered results, this approach returns more relevant results while still optimizing for search performance and computational costs.
```